external/fv3fit/fv3fit/emulation/thermobasis/loss.py

```diff
@@ -0,0 +1,151 @@
+import dataclasses
+from typing import Any, List, Tuple
+import tensorflow as tf
+from fv3fit.emulation.thermobasis.thermo import ThermoBasis
+from fv3fit.emulation.thermo import relative_humidity, specific_humidity_from_rh
+
+
+def rh_loss_info(truth_rh, pred_rh, level):
+    loss_rh = tf.reduce_mean(tf.losses.mean_squared_error(truth_rh, pred_rh))
+    return {
+        f"relative_humidity_mse/level_{level}": loss_rh.numpy() * (86400 / 900) ** 2
+    }
+
+
+def q_loss_info(truth_q, pred_q, level):
+    loss_q = tf.reduce_mean(tf.losses.mean_squared_error(truth_q, pred_q))
+    return {
+        f"loss/variable_3/level_{level}": loss_q.numpy() * (1000 * 86400 / 900) ** 2
+    }
+
+
+@dataclasses.dataclass
+class QVLoss:
+    """Loss function for predicting specific humidity at a single level
+    
+    Attributes:
+        level: the level to predict
+        scale: the typical order of the loss function
+    """
+
+    level: int
+    scale: float = 1.0
+
+    def loss(self, pred: tf.Tensor, out: ThermoBasis):
+        truth_q = select_level(out.q, self.level)
+        loss = tf.reduce_mean(tf.losses.mean_squared_error(truth_q, pred))
+
+        pred_rh = relative_humidity(
+            select_level(out.T, self.level), pred, select_level(out.rho, self.level),
+        )
+        truth_rh = select_level(out.rh, self.level)
+
+        return (
+            loss / self.scale,
+            {
+                **q_loss_info(truth_q, pred, self.level),
+                **rh_loss_info(truth_rh, pred_rh, self.level),
+            },
+        )
+
+
+@dataclasses.dataclass
+class RHLoss:
+    """Loss function for predicting relative humidity at a single level
+
+    Attributes:
+        variable: the variable to target, defaults to all levels of u,v,t,q
+        level: the level to predict
+        scale: the typical order of the loss function
+    """
+
+    level: int
+    scale: float = 1.0
+
+    def loss(self, pred_rh: tf.Tensor, out: ThermoBasis) -> Tuple[tf.Tensor, Any]:
+
+        pred_q = specific_humidity_from_rh(
+            select_level(out.T, self.level), pred_rh, select_level(out.rho, self.level),
+        )
+
+        truth_q = select_level(out.q, self.level)
+        truth_rh = select_level(out.rh, self.level)
+
+        loss_rh = tf.reduce_mean(tf.losses.mean_squared_error(truth_rh, pred_rh))
+
+        return (
+            loss_rh / self.scale,
+            {
+                f"loss": loss_rh.numpy(),
+                **q_loss_info(truth_q, pred_q, self.level),
+                **rh_loss_info(truth_rh, pred_rh, self.level),
+            },
+        )
+
+
+def select_level(arr: tf.Tensor, level) -> tf.Tensor:
+    return arr[:, level : level + 1]
+
+
+@dataclasses.dataclass
+class MultiVariableLoss:
+    """MSE loss function with manual weights for different variables
+
+    Attributes:
+        ?_weight: weight of the variable in the loss function.
+            Only used if name is None
+        levels: levels to save outputs at
```

Comment:
Is this saving outputs?

Reply:
No. This just adjusted the `info` outputs of `.loss`. See a831fded.